src/table/utils/selections-utils.ts

```diff
@@ -144,35 +161,91 @@ export const getSelectedRows = (
   return { ...selectedRows };
 };
 
+export const getMultiSelectedRows = (
+  selectedRows: Record<string, number>,
+  allRows: Row[],
+  evt: React.KeyboardEvent | React.MouseEvent,
+  cell: Cell,
+  firstCell?: Cell
+): Record<string, number> => {
+  const newSelectedRows = { ...selectedRows };
+
+  if ('key' in evt && evt.shiftKey && evt.key.includes('Arrow')) {
+    newSelectedRows[cell.qElemNumber] = cell.rowIdx;
+    // add the next or previous cell to selectedRows, based on which arrow is pressed
+    if (evt.key === 'ArrowDown') {
+      newSelectedRows[cell.nextQElemNumber as number] = cell.rowIdx + 1;
+    } else {
+      newSelectedRows[cell.prevQElemNumber as number] = cell.rowIdx - 1;
+    }
+    return newSelectedRows;
+  }
+
+  if (!firstCell) return selectedRows;
+  const lowestCellIdx = Math.min(firstCell.rawRowIdx, cell.rawRowIdx);
+  const highestCellIdx = Math.max(firstCell.rawRowIdx, cell.rawRowIdx);
+
+  for (let idx = lowestCellIdx; idx <= highestCellIdx; idx++) {
+    const selectedCell = allRows[idx][`col-${firstCell.rawColIdx}`] as Cell;
+    newSelectedRows[selectedCell.qElemNumber] = selectedCell.rowIdx;
+  }
```

Comment:
suggestions of smarter ways of doing this are appreciated. If we have to loop all cells from `firstCell` to `cell`, I think this is the fastest way

Reply:
LGTM honestly, trusty old for-loop does the trick

Reply:
that is fine, you **have to** iterate through selected ones and you are already using O(n) solution with accessing cell data with `allRows[idx][...]`
do you have access to first + last selected cell idx here? what is cell? 

I was thinking if you have start and end indexes, then you should not loop over all rows right? even the selection comes in different chunks (skip 10 rows, select 5 row, skip 15 rows, select 6 rows, ...)